helix-term/src/ui/editor.rs

```diff
@@ -208,6 +208,23 @@ impl EditorView {
             decorations,
         );
 
+        // Render ghost text at cursor position
+        if let Some(completion) = doc.inline_completions.current() {
+            if let Some(cursor_pos) = editor.cursor_cache.get(view, doc) {
```

Comment:
Thanks for starting on this! To echo the comment on the linked issue, rather than modifying the rendering code we should be leveraging a mix of LineAnnotation (to modify the current line) and Decoration to display the virtual lines:

https://github.com/helix-editor/helix/issues/13039#issuecomment-2726028429